### PR TITLE
Automated script to install this WONDERFUL tool!

### PR DESCRIPTION
**Pull request by kz6fittycent**
First, THANK YOU for creating this awesome tool to enable switchable graphics in Solus. I LOVE it!

I've created a script that will automate the installation for users and it supports the main 3 DE's (MATE, Budgie and Gnome). 

I've tested it several times and run it on my own machine(s). 

I don't know if you're still supporting this repo but I'd like to submit my PR anyway. Otherwise, I'm happy to keep things going as I can.

### CODE REVIEW

**Comment by ghost** [comment]
Why do you put `99-nvidia.conf` in `${GDM}` for Budgie desktop (it uses LightDM)?
You can create a function for things you repeat (`#SYSTEM PREP`).
There is no `else` statement which is bad because you need to say it if your script does not recognize the DE.
There is no check for su privilege.
You should create your variables using `readonly`.
Why so much sleep?

File: nvidia-optimus-installer.sh [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    # SYSTEM PREP
+    echo ""
+    echo "Getting things ready..."
+    echo ""
+    sudo mkdir -p ${MOD}
+    echo "blacklist nouveau" | sudo tee ${MOD}/blacklist-nouveau.conf
+    echo ""
+    echo "Installing necessary applications..."
+    echo ""
+    sudo eopkg it pciutils
```
Comment (ghost) [inline nvidia-optimus-installer.sh:61]: This part is repeated three times

File: nvidia-optimus-installer.sh [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+fi
```
Comment (ghost) [inline nvidia-optimus-installer.sh:109]: No else statement

File: nvidia-optimus-installer.sh (hunk 1 above)
Comment (ghost) [reply nvidia-optimus-installer.sh:61]: 1/3

File: nvidia-optimus-installer.sh [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    # SYSTEM PREP
+    echo ""
+    echo "Getting things ready..."
+    echo ""
+    sudo mkdir -p ${MOD}
+    echo "blacklist nouveau" | sudo tee ${MOD}/blacklist-nouveau.conf
+    echo ""
+    echo "Installing necessary applications..."
+    echo ""
+    sudo eopkg it pciutils
```
Comment (ghost) [inline nvidia-optimus-installer.sh:84]: 2/3

File: nvidia-optimus-installer.sh [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
```
Comment (ghost) [inline nvidia-optimus-installer.sh:108]: 3/3

File: nvidia-optimus-installer.sh [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+# VARIABLES
+####################################
+LDM=/etc/lightdm/lightdm.conf.d    # OLD DIRECTORY - USER CONFIRM
+GDM=/etc/gdm
+MOD=/etc/modprobe.d
+BIN=/usr/bin
+OPT=nvidia-optimus-manager
+DIR=`pwd`
+SYSD=nvidia-optimus-autoconfig.service
```
Comment (ghost) [inline nvidia-optimus-installer.sh:13]: Should be readonly

**Review by ghost — CHANGES_REQUESTED** [review]
I already posted a comment but I discovered this feature ^^

**Comment by kz6fittycent** [comment]
> Why do you put `99-nvidia.conf` in `${GDM}` for Budgie desktop (it uses LightDM)?
> You can create a function for things you repeat (`#SYSTEM PREP`).
> There is no `else` statement which is bad because you need to say it if your script does not recognize the DE.
> There is no check for su privilege.
> You should create your variables using `readonly`.
> Why so much sleep?

I think you should double-check the defaults for Budgie again using the commands listed in the original README. Several users have confirmed these settings for Budgie across several machines. 

As far as "repeating" things, it really doesn't matter, does it? 

Read only isn't necessary. There's no real security issue here. It's a ONE time script that plops files in place for the user. 

Why don't you redo the script the way you like it? I did a PR out of courtesy.

**Comment by VoltaVX** [comment]
@kz6fittycent I don't think Franck said those things out of bad faith or to be an ass, but wanted to keep some level of code-cleanness on his project. I'm sure he appreciates your contribution and everything, I mean, he did go out of his way to give you some advice on stuff ya can fix/clean

**Pull request closed without merging**